src/main/java/Poppy.java

```diff
@@ -0,0 +1,155 @@
+                list[counter] = deadline;
+                counter++;
+                break;
+            case "event":
+                String[] eventstring = str[1].split( "/from", 2);
+                Events event = new Events(eventstring[0], eventstring[1]);
+                list[counter] = event;
+                counter++;
+                break;
+            default:
+                list[counter] = new Task(input);
+                counter++;
+                echo(list[counter - 1]);
+            }
+            input =sc.nextLine();
+        }
+        System.out.println("Bye. Hope to see you again soon!");
+        System.exit(0);
+    }
+
+    public static void echo(Task task) {
+        System.out.println("added: " + task.description);
+    }
+
+    public static void markAsDone(Task[] list, String[] str) {
+        int index = Integer.parseInt(str[1]) - 1;
+        if (list[index]== null){
+            System.out.println("Sorry, list number does not exist");
+        }else {
+            list[index].markAsDone();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(list[index].taskType() + " [" + list[index].getStatusIcon() + "] " + list[index].description);
+        }
+    }
+
+    public static void showList(Task[] list){
+        System.out.println("Here are the tasks in your list: ");
+        for (int i = 0; i < list.length; i++) {
+            if (list[i] != null) {
+                System.out.println((i + 1) + ". " +list[i].taskType() +" [" + list[i].getStatusIcon() + "] " + list[i].description);
+            }
+        }
+    }
+
+    public static void markAsNotDone(Task[] list, String[] str) {
+        int index = Integer.parseInt(str[1])- 1;
+        if (list[index]== null){
+            System.out.println("Sorry, list number does not exist");
+        }else {
+            list[index].markAsNotDone();
+            System.out.println("OK, I've marked this task as not done yet:");
+            System.out.println(list[index].taskType()+ " [" + list[index].getStatusIcon() + "] " + list[index].description);
+        }
+    }
+
+    public static class Task {
+        protected String description;
+        protected boolean isDone;
+
+        public Task(String description) {
+            this.description = description;
+            this.isDone = false;
+        }
+
+        public String getStatusIcon() {
+            return (isDone ? "X" : " "); // mark done task with X
+        }
+
+        public void markAsDone() {
+            this.isDone = true;
+        }
+
+        public void markAsNotDone() {
+            this.isDone = false;
+        }
+
+        public String taskType(){
+            return "[]";
+        }
+        //...
+    }
+    public static class ToDo extends Task{
+        public ToDo (String description){
+            super(description);
+            System.out.println("[T]" + " [" + super.getStatusIcon()+ "] " + description);
```

Comment:
I think it would be better to standardise the spacings.. especially between the plus sign